homeassistant/components/http/web_runner.py

```diff
@@ -0,0 +1,71 @@
+"""HomeAssistant specific aiohttp Site."""
+import asyncio
+from ssl import SSLContext
+from typing import List, Optional, Union
+
+from aiohttp import web
+from yarl import URL
+
+
+class HomeAssistantTCPSite(web.BaseSite):
+    """HomeAssistant specific aiohttp Site.
+
+    Vanilla TCPSite accepts only str as host. However, the underlying asyncio's
+    create_server() implementation does take a list of strings to bind to multiple
+    host IP's. To support multiple server_host entries (e.g. to enable dual-stack
+    explicitly), we would like to pass an array of strings. Bring our own
+    implementation inspired by TCPSite.
+    """
+
+    __slots__ = ("_host", "_port", "_reuse_address", "_reuse_port", "_hosturl")
+
+    def __init__(
+        self,
+        runner: "web.BaseRunner",
+        host: Union[str, List[str]],
+        port: int,
+        *,
+        shutdown_timeout: float = 60.0,
+        ssl_context: Optional[SSLContext] = None,
+        backlog: int = 128,
+        reuse_address: Optional[bool] = None,
+        reuse_port: Optional[bool] = None,
+    ) -> None:  # noqa: D107
+        super().__init__(
+            runner,
+            shutdown_timeout=shutdown_timeout,
+            ssl_context=ssl_context,
+            backlog=backlog,
+        )
+
+        if isinstance(host, list):
+            if len(host) < 1:
+                raise RuntimeError("Empty list not allowed")
+            self._hosturl = host[0]
+        else:
+            self._hosturl = host
+
+        self._host = host
+        self._port = port
+        self._reuse_address = reuse_address
+        self._reuse_port = reuse_port
+
+    @property
+    def name(self) -> str:  # noqa: D102
+        scheme = "https" if self._ssl_context else "http"
+        return str(URL.build(scheme=scheme, host=self._hosturl, port=self._port))
```

Comment:
I think that we can avoid saving `self._hosturl`. We just need to fix the name here.

```suggestion
        host = self._host[0] if self._host else "0.0.0.0"
        return str(URL.build(scheme=scheme, host=host, port=self._port))
```